packages/core/src/lib/markdown-it/highlight/helper.js

```diff
@@ -7,6 +7,52 @@ function splitCodeAndIndentation(codeStr) {
   return [indents, content];
 }
 
+const BOUNDARY_TYPE = {
+  Start: Symbol('start'),
+  End: Symbol('end'),
+};
+
+// Simplifies multiple bounds applied on a single line to an array of disjointed bounds
+// boundaryCollection:
+// e.g [{index: 1, type: BOUNDARY_TYPE.Start},
+//      {index:3, type: BOUNDARY_TYPE.End},
+//      {index: 5, type: BOUNDARY_TYPE.Start},
+//      {index: 7, type: BOUNDARY_TYPE.End}]
+function collateAllIntervals(boundaryCollection) {
+  let startCount = 0;
+  let endCount = 0;
+  let boundStart;
+  let boundEnd;
+  const output = [];
+  boundaryCollection.sort((boundaryA, boundaryB) => boundaryA.index - boundaryB.index);
```

Comment:
Do we need to tie break here to prioritise `BOUNDARY_TYPE.Start` (or `END`)?

Reply:
Could add this scenario in `testCodeBlocks.md` too if there's isn't one already.

Reply:
Hmm, I suppose prioritising start will make it more concise?
Idea is that prioritising end might cause a currently opened interval to be collected if say the number of starts is just one more than number of ends. And result in a new open interval to begin at the same index when both can technically be merged since they are continuous.

Thanks for the suggestion.

Reply:
I will add the test case as well.